stdlib/collections/__init__.pyi

```diff
@@ -137,8 +137,10 @@ class UserList(MutableSequence[_T]):
     def copy(self) -> Self: ...
     def __copy__(self) -> Self: ...
     def count(self, item: _T) -> int: ...
-    # All arguments are passed to `list.index` at runtime, so the signature should be kept in line with `list.index`.
-    def index(self, item: _T, __start: SupportsIndex = 0, __stop: SupportsIndex = sys.maxsize) -> int: ...
+    # The runtime signature is "item, *args", and the arguments are then passed
+    # to `list.index`. In order to give more precise types, we pretend that the
+    # `item` argument is positional-only.
```

Comment:
This is a bit unfortunate because it disallows a legal call `lst.index(item="x")`. However, it's unlikely anyone would write that, and I don't see a better alternative that preserves type safety.